docs/UsersGuide.asciidoc

```diff
@@ -193,28 +193,36 @@ by different means:
 
 
 === Variable expansion
+Any job setting can refer to another variable using this syntax: `%NAME%`. When
+the test job is created, the string will be substituted with the value of the
+specified variable at that time.
```

Comment:
```suggestion
openQA supports variable expansion, surrounding a test variable with percentage signs (`%`), e.g `%NAME%`.

Variable expansion can happen only at two stages:

1. When the job is created the string will be substituted with the value of the
specified variable at that time. This is useful to handle static information, like a directory or a repository.
1. Before the download state of the worker, making it a runtime value, that will not be shown in the openQA webUI, but is shown in the logs. i.e runtime provisioning of ephemeral values, or using different values defined at the worker level.
```

Reply:
I would go as far as to say that also protected values like `SECRET` or `_PASSWORD` would also be a great case for https://progress.opensuse.org/issues/109019 and https://progress.opensuse.org/issues/105624

but for now that's a random crazy idea

Reply:
I wouldn't make this so long and avoid generic expressions like "the string". However, maybe I'll incorporate some parts of it.